akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -102,6 +103,30 @@ import scala.concurrent.duration._
  */
 @InternalApi private[typed] object ActorContextAdapter {
 
+  // just to get access to the class context
+  private final class TrickySecurityManager extends SecurityManager {
+    def getClassStack: Array[Class[_]] = getClassContext
+  }
+
+  private def getLoggerClass(system: ExtendedActorSystem): Class[_] = {
+    // FIXME use stack walker API when we no longer need to support Java 8
+    val trace = new TrickySecurityManager().getClassStack
+    var suitableClass: OptionVal[Class[_]] = OptionVal.None
+    var idx = 2 // skip this method and ctx.log right away
+    while (suitableClass.isEmpty && idx < trace.length) {
+      val clazz = trace(idx)
+      val name = clazz.getName
+      if (!name.startsWith("scala.runtime") && !name.startsWith("akka.actor.typed.internal"))
```

Comment:
what happens for the actors we have here, such as `LocalReceptionist`?

Reply:
Hmm. I wonder if we should just select class explicitly for those, or be more specific with this filter perhaps all the way down to `Behaviors`

Reply:
I agree that we should define it explicitly for our actors, but there is a risk that we forget and it will not be noticed

Reply:
Set it explicitly with the setLoggerClass for the receptionists now.